src/uwtools/config/formats/base.py

```diff
@@ -158,43 +181,35 @@ def _parse_include(self, ref_dict: Optional[dict] = None) -> None:
 
     # Public methods
 
-    def compare_config(self, dict1: dict, dict2: Optional[dict] = None) -> bool:
+    @abstractmethod
+    def as_dict(self) -> dict:
+        """
+        Returns a pure dict version of the config.
+        """
```

Comment:
This is mainly to support `NMLConfig`, whose `.data` attribute can be an `f90nml` `Namelist` object which, although `dict`-like, is not actually a `dict` and runs into problems in `yaml.dump()`. `NMLConfig` now implements `as_dict()` in such a way (see below) that it returns a pure-`dict` object. The other config types just return their `.data` attribute, which is already a `dict`.